pkg/network/common/dns.go

```diff
@@ -154,8 +154,13 @@ func (d *DNS) getIPsAndMinTTL(domain string) ([]net.IP, time.Duration, error) {
 				case *dns.A:
 					ips = append(ips, t.A)
 
-					if minTTL == 0 || t.Hdr.Ttl < minTTL {
-						minTTL = t.Hdr.Ttl
+					// We could potentially get zero TTL as we try to query the DNS
+					// server at the end of the TTL value for the domain.
+					// Adding 1 sec will ensure non zero minTTL
+					ttl := t.Hdr.Ttl + 1
+
+					if minTTL == 0 || ttl < minTTL {
```

Comment:
Can't this just be:
if ttl < minTTL {

Why do we care if minTTL == 0

Reply:
We were overloading this field to know whether this field is set or not otherwise we don't care about the value of minTTL. Let me fix this one.